ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/infrastructure/FutureAssertions.scala

```diff
@@ -125,24 +126,26 @@ object FutureAssertions {
       data: Seq[T]
   )(
       testCase: T => Future[Unit]
-  )(implicit ec: ExecutionContext, loggingContext: LoggingContext): Future[Seq[Unit]] = Future
+  )(implicit ec: ExecutionContext, loggingContext: LoggingContext): Future[Unit] = Future
     .traverse(data)(input =>
       testCase(input).map(Right(_)).recover { case NonFatal(ex) =>
         Left(input -> ex)
       }
     )
     .map { results =>
-      val (failures, successes) = results.partitionMap(identity)
-      if (failures.nonEmpty) {
-        failures
-          .foreach(res => logger.error(s"Failed parallel test case for input ${res._1}", res._2))
-        throw ParallelTestFailureException(
-          s"Failed parallel test case. Failures: ${failures.length}. Success: ${successes.length}\nFailed inputs: ${failures
-            .map(_._1)
-            .mkString("[", ",", "]")}",
-          failures.last._2,
-        )
-      } else successes
+      discard {
+        val (failures, successes) = results.partitionMap(identity)
+        if (failures.nonEmpty) {
+          failures
+            .foreach(res => logger.error(s"Failed parallel test case for input ${res._1}", res._2))
+          throw ParallelTestFailureException(
+            s"Failed parallel test case. Failures: ${failures.length}. Success: ${successes.length}\nFailed inputs: ${failures
+              .map(_._1)
+              .mkString("[", ",", "]")}",
+            failures.last._2,
+          )
+        } else successes
```

Comment:
It looks like you are diverging in the `if` branch, so I think you can return unit or even remove the `else` branch altogether, as you are discarding `successes` anyway.